presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -367,6 +380,19 @@ private void freePartitions()
             // Remove out references to partitions to actually free memory
             Arrays.fill(partitions, null);
             lookupSourceSupplier = null;
+            closeCachedLookupSources();
+        }
+        finally {
+            rwLock.writeLock().unlock();
+        }
+    }
+
+    private void closeCachedLookupSources()
+    {
+        rwLock.writeLock().lock();
```

Comment:
looks like this is already called from within locks, so you might want to just mark it `@GuardedBy(rwLock)`

Reply:
Interesting, I'm not sure you can.  @martint, said that guarded by is not about exclusivity, but is about memory visibility. So from an `@GuardedBy` perspective, saying `rwLock` is good enough.... I agree with this, but in that case, I'd rather see the explicit locking.